Provide faster (DBAL-based) solution for resolution of regular nodes

robertlemke opened the pull request.

robertlemke committed:
Provide faster (DBAL-based) solution for resolution of regular nodes

kitsunet reviewed (approved):
Works, as long as there are no dimensions and solves the performance issue.

robertlemke committed:
Fall back to non-DBAL-solution for multiple dimenstion presets

The DBAL optimization does not support multiple dimensions yet, therefore
fall back to the original implementation if multiple dimensions presets
are configured.
